sql/create.go

```diff
@@ -142,6 +142,8 @@ func (p *planner) CreateIndex(n *parser.CreateIndex) (planNode, error) {
 		return nil, err
 	}
 
+	// Mark transaction as operating on the system DB.
+	p.txn.SetSystemDBTrigger()
```

Comment:
actually, I may have to keep the field in the transaction itself. If we end up going through the txn_coord_sender, I need to propagate the setting so that we end up with the proper trigger on the EndTransactionRequest.
Technically, the one I could get it of is the trigger itself and just look at the field in the transaction.